sycl/include/CL/sycl/intel/fpga_lsu.hpp

```diff
@@ -0,0 +1,116 @@
+//==-------------- fpga_lsu.hpp --- SYCL FPGA Reg Extensions ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#pragma once
+
+#include "fpga_utils.hpp"
+#include <CL/sycl/detail/defines.hpp>
+#include <CL/sycl/pointers.hpp>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace intel {
+constexpr unsigned BURST_COALESCE = 0x1;
```

Comment:
I changed the unsigned to `uint8_t`. That works because I only have 4 parameters for now, so no more than 4 bits are needed. However, the `int` can only be changed to `int32_t` because of how `GetValue` is written. All the parameters captured should have the same type because `GetValue` accepts only one type. I couldn't rewrite it otherwise :( Also, I want the type to be wide enough to be able to capture the size of the cache.